src/sinol_make/commands/run/__init__.py

```diff
@@ -202,41 +204,82 @@ def execute(self, execution):
-				if ": " in line:
-					(key, value) = line.split(": ")[:2]
-					result[key] = value
-		if "Time" in result.keys():
-			result["Time"] = self.parse_time(result["Time"])
-		if "Memory" in result.keys():
-			result["Memory"] = self.parse_memory(result["Memory"])
-		if code == 35072:
-			result["Status"] = "TL"
-		elif "Time" in result.keys() and result["Time"] > time_limit:
-			result["Status"] = "TL"
-		elif "Memory" in result.keys() and result["Memory"] > memory_limit:
-			result["Status"] = "ML"
-		elif "Status" not in result.keys():
-			result["Status"] = "RE"
-		elif result["Status"] == "OK":
-			if result["Time"] > time_limit:
+		if self.args.time_tool == 'oiejq':
+			command = "MEM_LIMIT=%sK MEASURE_MEM=true timeout -k %ds -s SIGKILL %ds %s %s <%s >%s 2>%s" \
+					% (math.ceil(memory_limit), hard_time_limit_in_s,
+						hard_time_limit_in_s, timetool_path,
+						executable, test, output_file, result_file)
+			timeout_exit_code = os.system(command)
+			result = {}
+			with open(result_file) as r:
+				for line in r:
+					line = line.strip()
+					if ": " in line:
+						(key, value) = line.split(": ")[:2]
+						result[key] = value
+			if "Time" in result.keys():
+				result["Time"] = self.parse_time(result["Time"])
+			if "Memory" in result.keys():
+				result["Memory"] = self.parse_memory(result["Memory"])
+
+			if timeout_exit_code == 35072:
+				result["Status"] = "TL"
+			elif "Time" in result.keys() and result["Time"] > time_limit:
+				result["Status"] = "TL"
+			elif "Memory" in result.keys() and result["Memory"] > memory_limit:
+				result["Status"] = "ML"
+			elif "Status" not in result.keys():
+				result["Status"] = "RE"
+			elif result["Status"] == "OK":
+				if result["Time"] > time_limit:
+					result["Status"] = "TL"
+				elif result["Memory"] > memory_limit:
+					result["Status"] = "ML"
+				elif os.system("diff -q -Z %s %s >/dev/null"
+							% (output_file, self.get_output_file(test))):
+					result["Status"] = "WA"
+			else:
+				result["Status"] = result["Status"][:2]
+
+			return result
+		elif self.args.time_tool == 'time':
+			if sys.platform == 'darwin':
+				command = 'gtimeout -k %ds %ds gtime -f "%%U\\n%%M\\n%%x" -o %s %s <%s >%s' \
+					% (hard_time_limit_in_s, hard_time_limit_in_s,
+						result_file, executable, test, output_file)
+			elif sys.platform == 'linux':
+				command = 'timeout -k %ds %ds time -f "%%U\\n%%M\\n%%x" -o %s %s <%s >%s' \
+					% ( hard_time_limit_in_s, hard_time_limit_in_s,
+						result_file, executable, test, output_file)
+			elif sys.platform == 'win32' or sys.platform == 'cygwin':
+				raise Exception("Measuring time with GNU time on Windows is not supported.")
+
+			timeout_exit_code = os.system(command)
+
+			result = {}
+			lines = open(result_file).readlines()
+			program_exit_code = None
+			if len(lines) == 3:
+				result["Time"] = round(float(lines[0].strip()) * 1000)
+				result["Memory"] = int(lines[1].strip())
+				program_exit_code = int(lines[2].strip())
```

Comment:
When doing such hacky parsing it's good to write a comment with an example output, so that I can see whether the code makes sense on an example. Also, the `time` output might differ for different operating systems or even different shells in the same operating system, so it might be beneficial to assert how the output looks like, or maybe both parse and assert at once.

Reply:
Some comments about output of time command: fbbc71e26f9a547237dbe255700ba132d5738eb1. I'm using time's -f flag which specifies the format.